src/NuGet.CommandLine.StagingExtensions/StageManagementResource.cs

```diff
@@ -107,6 +109,39 @@ public StageManagementResource(string stageServiceUri, HttpSource httpSource)
             return result;
         }
 
+        public async Task<IReadOnlyList<StageListView>> List(string apiKey, Common.ILogger log)
+        {
+            if (string.IsNullOrWhiteSpace(apiKey))
+            {
+                throw new ArgumentException(StagingResources.ApiKeyShouldNotBeEmpty, nameof(apiKey));
+            }
+
+            if (log == null)
+            {
+                throw new ArgumentNullException(nameof(log));
+            }
+
+            var result = await _httpSource.ProcessResponseAsync(
+                () =>
+                {
+                    var request = new HttpRequestMessage(HttpMethod.Get, new Uri(new Uri(_stageServiceUri), $"{StagePath}"));

```

Comment:
Is `_stageServiceUri` not already a URI? It should probably be renamed or this should be reduced to `new Uri(_stageServiceUri, $"{StagePath}")`.